react/week4/src/components/SearchForm.js

```diff
@@ -0,0 +1,26 @@
+import React,{useContext} from 'react';
+import{UserContext} from './FetchUser';
+
+const SearchForm = () => {
+
+const{query,setQuery} = useContext(UserContext);
+
+const UpdateUser=(e) =>{
+    e.preventDefault();
+    setQuery(e.target.value)
```

Comment:
This could problematic. It's good practice to delay the network requests when interacting with keystrokes. It's a technique called `debouncing`. Please read [this article](https://css-tricks.com/debouncing-throttling-explained-examples/) about what is it, how it works and how it differs from the `throttling` technique. It's basically what Niels said in [this](https://hackyourfuture-cph.slack.com/archives/CL198NB4G/p1582073569234800) slack post.

Reply:
will look on it , thanks a lot for reviewing Dan